xcube/core/new.py

```diff
@@ -118,12 +118,6 @@ def my_func(time: int, y: int, x: int) -> Union[bool, int, float]
     x_end = x_start + width * x_res
     y_end = y_start + height * y_res
 
-    if x_is_lon and x_start < -180. or x_end > 180.:
-        raise ValueError()
-
-    if y_is_lat and y_start < -90. or y_end > 90.:
-        raise ValueError()
-
```

Comment:
I am not quite convinced removing this is the best solution, at the very least the latitude check makes sense. In the case of the one failing test, the failure could be prevented by using the correct units (or deriving them from the provided crs).

EDIT: I just found that adding the correct units does not work, regardless of whether these lines are kept or not. There is some bug here.

Reply:
Okay. 

My main issue here was that a test was not working anymore if you set correct units, but when I had a closer look I found that this is due to the characteristics of the test, the acual code works well.